bleak/backends/client.py

```diff
@@ -42,9 +42,7 @@ def __init__(self, address_or_ble_device: Union[BLEDevice, str], **kwargs):
         else:
             self.address = address_or_ble_device
 
-        self.services = BleakGATTServiceCollection()
-
-        self._services_resolved = False
+        self._services: Optional[BleakGATTServiceCollection] = None
```

Comment:
> We have some users that have made custom backends, so changing services to a property here would be a breaking change for them. ~ @dlech

Do you think having `self.services` as `Optional` would be the same breaking change for them? In such case this PR is kinda useless - while testing paring I just saw that we have a redundant `_services_resolved` and I thought of removing it.

Reply:
No, I don't think it is breaking. If `self.services` is never `None`, it just won't trigger the new exception.